lib/bundle.js

```diff
@@ -20,158 +20,253 @@ var Builder = require('systemjs-builder');
-
-    // trace the starting module
-    var cfg = config.loader.getConfig();
-    cfg.baseURL = path.relative(process.cwd(), config.pjson.baseURL);
-
-    return systemBuilder.trace(moduleName, cfg);
-  })
-  .then(function(output) {
-    var traceTree = output.tree;
-    moduleName = output.moduleName;
-    var depCache = config.loader.depCache = config.loader.depCache || {};
-    for (var d in traceTree) {
-      var deps = traceTree[d].deps.map(function(dep) {
-        return traceTree[d].depMap[dep];
-      });
+    .then(function() {
+      moduleName = moduleName || config.loader.main;
+    })
+    .then(function() {
+      ui.log('info', 'Injecting the traced dependency tree for `' + moduleName + '`...');
 
-      if (deps.length)
-        depCache[d] = deps;
-    }
-  })
-  .then(config.save)
-  .then(function() {
-    ui.log('ok', 'Depenency tree injected');
-  })
-  .catch(function(e) {
-    ui.log('err', e.stack || e);
-  });
+      // trace the starting module
+      var cfg = config.loader.getConfig();
+      cfg.baseURL = path.relative(process.cwd(), config.pjson.baseURL);
+
+      return systemBuilder.trace(moduleName, cfg);
+    })
+    .then(function(output) {
+      var traceTree = output.tree;
+      moduleName = output.moduleName;
+      var depCache = config.loader.depCache = config.loader.depCache || {};
+      for (var d in traceTree) {
+        var deps = traceTree[d].deps.map(function(dep) {
+          return traceTree[d].depMap[dep];
+        });
+
+        if (deps.length)
+          depCache[d] = deps;
+      }
+    })
+    .then(config.save)
+    .then(function() {
+      ui.log('ok', 'Depenency tree injected');
+    })
+    .catch(function(e) {
+      ui.log('err', e.stack || e);
+    });
 };
 
 function extractOperations(args) {
+  var i;
   var operations = [];
 
-  for (var i = 1; i < args.length - 1; i = i + 2) {
-    var operator = args[i];
-    var moduleName = args[i + 1];
-
-    operations.push({
-      operator: operator,
-      moduleName: moduleName
+  var seq = expandGlob(args[0])
+    .then(function(moduleNames) {
+      pushOps(moduleNames, '+');
     });
+
+  for (i = 1; i < args.length - 1; i += 2) {
+    seq = (function(name, operator) {
+      return seq.then(function() {
+        return expandGlob(name)
+          .then(function(moduleNames) {
+            pushOps(moduleNames, operator)
+          });
+      });
+    })(args[i + 1], args[i]);
   }
 
-  return operations;
+  seq = seq.then(function() {
+    return operations;
+  });
+
+  return seq;
+
+  function pushOps(moduleNames, operator) {
+    moduleNames.forEach(function(mn) {
+      operations.push({
+        moduleName: mn,
+        operator: operator
+      });
+    });
+  }
 }
 
 function extractBundleName(fileName) {
   return path.relative(config.pjson.baseURL, fileName.replace(/\.js$/, '')).replace(/\\/g, '/');
 }
 
+function getModuleName(file) {
+  var baseURL = config.pjson.baseURL + '/';
```

Comment:
Only add the trailing `/` if it doesn't already exist. Also for windows support we probably want this to be `path.sep`?